core/coreunix/add.go

```diff
@@ -49,11 +49,14 @@ type Object struct {
 
 // NewAdder Returns a new Adder used for a file add operation.
 func NewAdder(ctx context.Context, p pin.Pinner, bs bstore.GCBlockstore, ds ipld.DAGService) (*Adder, error) {
+	bufferedDS := ipld.NewBufferedDAG(ctx, ds)
```

Comment:
I'll probably refactor this to happen on the coreapi layer (next to offline/sessions stuff), but this is good enough for now